build-system/tasks/check-links.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var argv = require('minimist')(process.argv.slice(2));
+var chalk = require('chalk');
+var child_process = require('child_process');
+var fs = require('fs-extra');
+var gulp = require('gulp-help')(require('gulp'));
+var markdownLinkCheck = require('markdown-link-check');
+var path = require('path');
+var util = require('gulp-util');
+
+
+/**
+ * Executes the provided command; terminates this program in case of failure.
+ * Copied from pr-check.js.
+ * TODO(rsimha-amp): Refactor this into a shared library. Issue #9038.
+ *
+ * @param {string} cmd
+ */
+function execOrDie(cmd) {
+  var p =
+      child_process.spawnSync('/bin/sh', ['-c', cmd], {'stdio': 'inherit'});
+  if (p.status != 0) {
+    console/*OK*/.log('\nExiting due to failing command: ' + cmd);
+    process.exit(p.status)
+  }
+}
+
+/**
+ * Checks for dead links in the list of files.
+ */
+function checkLinks() {
+  var files = '';
+  if (argv.files) {
+    files = argv.files;
+  } else {
+    console./*OK*/error(util.colors.red(
+        'Must specify a list of files via --files'));
+    process.exit(1);
+  }
+  var markdownFiles = files.split(',');
+  markdownFiles.forEach(function(markdownFile) {
+    util.log('Checking links in', util.colors.magenta(files), '...');
+    var filteredMarkdownFile = filterLocalhostLinks(markdownFile);
+    runLinkChecker(filteredMarkdownFile);
```

Comment:
see my comment below, if we go with that approach this becomes:

runLinkChecker(markdownFile);